R/visr.R

```diff
@@ -560,3 +562,66 @@ visr.tidycuminc <- function(x = NULL
   gg
 }
 
+
+.construct_strata_label <- function(x, sep = ", ") {
+  tryCatch({
+    if (inherits(x, "survfit") && is.null(x$strata_lbl)) {
+      strata_label <- " "
+    }
+    else if (inherits(x, "survfit")) {
+      strata_label <- unlist(x$strata_lbl) %>% paste(collapse = ", ")
+    }
+    if (inherits(x, "tidycuminc")) {
+      data<- x$data
+    }
+
+
+
+    # get the original data frame
+    if (inherits(x, "survfit")) {
+      data <- as.list(x$call)[["data"]] %>% eval()
```

Comment:
If we have the logic of strata labels in estimate_xx it would make it easier. You would do all the heavy lifting in a spot where all the information is available. Using this logic, you wont need to rebuild what you originally had.

The disadvantage is when we are passing a foreign survift object to visr(). The labels wont be adjusted.

Do we sacrifice efficiency or compatibility?

Reply:
I am not sure what happened...I thought I had deleted these lines. Anyway, they are now removed!